src/web/src/app.tsx

```diff
@@ -0,0 +1,362 @@
+  list: string[] | null | undefined;
+}): JSX.Element {
+  if (props.list == null || props.list.length === 0) return <></>;
+  return (
+    <ul className="study-meta">
+      <span>{props.caption}: </span>
+      {props.list.map((c) => (
+        <li key={c}>{c}</li>
+      ))}
+    </ul>
+  );
+}
+
+export function IncludeExcludeList(props: {
+  caption: string;
+  include: string[] | null | undefined;
+  exclude: string[] | null | undefined;
+}): JSX.Element {
+  return (
+    <>
+      <PropertyList caption={props.caption} list={props.include} />
+      <PropertyList caption={'Exclude' + props.caption} list={props.exclude} />
+    </>
+  );
+}
+
+export function StudyItem(props: {
+  study: StudyModel;
+  filter: StudyFilter;
+}): JSX.Element {
+  const filter = props.study.filter();
+  return (
+    <div className="card mb-3">
+      <div className="card-header">
+        <a target="_blank" href={props.study.getConfigUrl()} rel="noreferrer">
+          {props.study.name()}
+        </a>
+      </div>
+      <div className="card-body">
+        <ul className="list-group list-group-flush">
+          {props.study.experiments(props.filter).map((e) => (
+            <ExperimentItem key={e.name()} exp={e} />
+          ))}
+        </ul>
+      </div>
+      <div className="card-footer">
+        <PropertyList caption="Channels" list={props.study.channels()} />
+        <PropertyList caption="Platforms" list={props.study.platforms()} />
+        <IncludeExcludeList
+          caption="Countries"
+          include={filter?.country}
+          exclude={filter?.exclude_country}
+        />
+        <IncludeExcludeList
+          caption="Hardware class"
+          include={filter?.hardware_class}
+          exclude={filter?.exclude_hardware_class}
+        />
+      </div>
+    </div>
+  );
+}
+
+function getCurrentSeedType(searchParams: URLSearchParams): SeedType {
+  return (
+    stringToSeedType(searchParams.get('seed') ?? 'PRODUCTION') ??
+    SeedType.PRODUCTION
+  );
+}
+
+export function NavItem(props: { type: SeedType }): JSX.Element {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const currentSeed = getCurrentSeedType(searchParams);
+  const className =
+    (props.type === currentSeed ? 'active ' : '') + 'nav-item nav-link btn-sm';
+  const handleClick = (): void => {
+    setSearchParams((prev) => {
+      prev.set('seed', SeedType[props.type]);
+      prev.delete('search');
+      return prev;
+    });
+  };
+  return (
+    <a onClick={handleClick} className={className}>
+      {SeedType[props.type]}
+    </a>
+  );
+}
+
+export function FilterCheckbox(props: {
+  title: string;
+  htmlName: string;
+  checked: boolean;
+  toggle: () => void;
+}): JSX.Element {
+  return (
+    <div className="filter">
+      <input
+        type="checkbox"
+        id={props.htmlName}
+        checked={props.checked}
+        onChange={props.toggle}
+      />
+      <label htmlFor={props.htmlName}>{props.title}</label>
```

Comment:
tip: You don't need to specify an `id` or an `htmlFor` if you wrap the input inside the label, e.g.

```tsx
<label>
  <input ... />
  {props.title}
</label>
```

Reply:
Thanks, done